### PR TITLE
fix(singular): undefined properties object for track

### PR DESCRIPTION
**Pull request by sanpj2292**
Signed-off-by: Sai Sankeerth <sanpj2292@github.com>

## Description of the change

https://www.notion.so/rudderstacks/Singular-Generic-Transformation-Error-79e1958c52424133b1240935072fe03e

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by yashasvibajpai — DISMISSED** [review]

File: src/v0/destinations/singular/transform.js [hunk 1]
```diff
@@ -23,13 +23,15 @@ const responseBuilderSimple = (message, { Config }) => {
...
-    // If we have an event where we have an array of Products, example Order Completed
-    // We will convert the event to revenue events
-    if (products && Array.isArray(products)) {
-      return generateRevenuePayloadArray(products, payload, Config, eventAttributes);
-    }
+  // If we have an event where we have an array of Products, example Order Completed
+  // We will convert the event to revenue events
+  if (!sessionEvent && Array.isArray(message?.properties?.products)) {
+    return generateRevenuePayloadArray(
+      message?.properties?.products,
```
Comment (saikumarrs) [inline src/v0/destinations/singular/transform.js:30]: No need of optional chaining here. Once, addressed resolve the comment yourself.

**Review by saikumarrs — APPROVED** [review]

File: test/__tests__/data/singular_router_input.json [hunk 2]
```diff
@@ -340,5 +340,307 @@
...
+            "amp",
+            "cloud",
+            "warehouse"
+          ],
+          "saveDestinationResponse": true
+        },
+        "configSchema": {},
+        "responseRules": {},
+        "options": null,
+        "uiConfig": [
```
Comment (krishna2020) [inline test/__tests__/data/singular_router_input.json:489]: Remove `uiConfig` section in both input & output --> this is redundant info in the test cases

File: src/v0/destinations/singular/transform.js (hunk 1 above)
Comment (sanpj2292) [reply src/v0/destinations/singular/transform.js:30]: If we just do `message.properties.products`, we would get an error like `Cannot access "products" of undefined`. Hence opetional chaining is necessary

Comment (saikumarrs) [reply src/v0/destinations/singular/transform.js:30]: It's already validated in the above if condition statement (which already uses optional chaining). Hence, unnecessary.

Comment (sanpj2292) [reply src/v0/destinations/singular/transform.js]: Done 🙏🏼

File: test/__tests__/data/singular_router_input.json (hunk 2 above)
Comment (sanpj2292) [reply test/__tests__/data/singular_router_input.json]: Done

**Review by saikumarrs — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**